### PR TITLE
[Merged by Bors] - fix: use Name.isInternal' in LibrarySearch

### PR DESCRIPTION
**Pull request by dwrensha**
Uses `isInternal'` it in `library_search` to ignore generated lemmas with names such as `proof_1`.

Before this change, in the first testcase https://github.com/leanprover-community/mathlib4/blob/5d55252d5e02341c6cf996e8d658016c104fb423/test/librarySearch.lean#L5
`library_search` suggests:
```lean
exact
  ConditionallyCompleteLattice.copy.proof_1 ConditionallyCompleteLinearOrder.toConditionallyCompleteLattice Nat.le rfl
    (Nat.succ x)
```
After this change, it suggests:
```lean
exact Nat.lt.base x
```

https://github.com/leanprover-community/mathlib4/pull/2348 is another example of where these `proof_1` terms can appear.

### CODE REVIEW

**Comment by dwrensha** [comment]
Hm... I wonder if we should instead change core Lean so that it prefixes these generated constants with an underscore, allowing the existing `isInternal` function catch them?

**Comment by dwrensha** [comment]
A proposal for fixing this upstream: https://github.com/leanprover/lean4/pull/2112

File: Mathlib/Lean/Data/Name.lean [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+open Lean
+
+/-- Lean 4 makes declarations which are technically not internal
+(that is, head string does not start with `_`) but which sometimes should
+be treated as such. For example, the `to_additive` attribute needs to
+transform `proof_1` constants generated by `Lean.Meta.mkAuxDefinitionFor`.
+This might be better fixed in core, but until then, this method can act
+as a polyfill.
+Note: this declaration also occurs as `shouldIgnore` in the Lean 4 file `test/lean/run/printDecls`.
+-/
```
Comment (eric-wieser) [inline Mathlib/Lean/Data/Name.lean:24]: Can you add a reference to the upstream issue in a comment?

**Review by eric-wieser — APPROVED** [review]
LGTM. I don't think we need to wait for an upstream fix for this; we can always just delete the function once the fix is in.

I do wonder if we want a flag to enable searching for generated declarations too (if the other search falls through), but that doesn't need to be in this PR.

File: Mathlib/Lean/Data/Name.lean (hunk 1 above)
Comment (dwrensha) [reply Mathlib/Lean/Data/Name.lean]: Done in https://github.com/leanprover-community/mathlib4/pull/2349/commits/aa1f1844b634bf509934c3b89fb906924b7de1de

**Comment by fpvandoorn** [comment]
I also move `isInternal` in #2045, and it should be noted that the linter framework uses their own (more expansive) check for automatically generated declarations (`isAutoDecl`), and @kmill mentions `shouldIgnoreDecl` in #2045 (though I couldn't find that one), so we should probably do some uniformization as well, though not all of these declarations have the same function.

Further, I think we should absolutely not try any internal declarations in `library_search`, and don't see a point in an option to allow using them.

**Comment by kmill** [comment]
Here's the original source of the definition I mentioned in #2045 (that particular one, `shouldIgnoreDecl`, is the version I was using in one of my own projects): https://github.com/leanprover/doc-gen4/blob/main/DocGen4/Process/DocInfo.lean#L120

**Comment by fpvandoorn** [comment]
bors d+

Please merge master after #2045 reaches master. You can choose whether you use your or my placement of the declaration. I added the first paragraph of your version of the docs, but I might have forgotten the second paragraph.

**Pull request closed without merging**